### PR TITLE
Orders improvements

### PR DESCRIPTION
**Pull request by capsert**

### CODE REVIEW

File: examples/13-handle-order-status-change.py [hunk 1]
```diff
@@ -27,34 +30,18 @@ def main():
         #
         # See: https://docs.mollie.com/reference/v2/orders-api/get-order
         #
+        if 'id' not in flask.request.form:
+            flask.abort(404, 'Unknown payment id')
```
Comment (whyscream) [inline examples/13-handle-order-status-change.py]: There are no payments here

File: examples/17-order-return-page.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from mollie.api.client import Client
+
+
+def main():
+    api_key = os.environ.get('MOLLIE_API_KEY', 'test_test')
+    mollie_client = Client()
+    mollie_client.set_api_key(api_key)
+
+    if 'my_webshop_id' not in flask.request.args:
+        flask.abort(404, 'Unknown order_id')
```
Comment (whyscream) [inline examples/17-order-return-page.py]: You weren't looking for a order_id in the first place.

File: examples/2-webhook-verification.py [hunk 3]
```diff
@@ -31,33 +31,13 @@ def main():
...
         payment = mollie_client.payments.get(payment_id)
-        order_id = payment.metadata['order_id']
+        my_webshop_id = payment.metadata['my_webshop_id']
 
         #
         # Update the order in the database.
         #
-        database_write(order_id, payment.status)
-
-        if payment.is_paid():
```
Comment (whyscream) [inline examples/2-webhook-verification.py]: The webhook is a great place for handling changes after changes take place at Mollie. The code below serves as an example for what you can do when you receive a webhook reques: "At this point you'd probably want to start the process of delivering the product to the customer". Why are you leaving it out?

File: examples/4-ideal-payment.py [hunk 4]
```diff
@@ -68,9 +68,9 @@ def main():
             },
             'description': 'My first API payment',
             'webhookUrl': '{root}2-webhook_verification'.format(root=flask.request.url_root),
-            'redirectUrl': '{root}3-return-page?order_id={id}'.format(root=flask.request.url_root, id=order_id),
+            'redirectUrl': '{root}3-return-page?order_id={id}'.format(root=flask.request.url_root, id=my_webshop_id),
```
Comment (whyscream) [inline examples/4-ideal-payment.py]: the parameter in the url is still named 'order_id'

File: examples/9-create-customer-payment.py [hunk 5]
```diff
@@ -62,13 +62,13 @@ def main():
             },
             'description': 'My first API payment',
             'webhookUrl': '{root}2-webhook_verification'.format(root=flask.request.url_root),
-            'redirectUrl': '{root}3-return-page?order_id={id}'.format(root=flask.request.url_root, id=order_id),
+            'redirectUrl': '{root}3-return-page?order_id={id}'.format(root=flask.request.url_root, id=my_webshop_id),
```
Comment (whyscream) [inline examples/9-create-customer-payment.py]: parameter name in url hasn't changed

File: examples/app.py [hunk 6]
```diff
@@ -42,20 +44,23 @@ def run_example(example=None):
     app.debug = True
     app.run()
 
+
 #
 # NOTE: This example uses a plain txt file as a "database". Please use a real database like MySQL in production.
```
Comment (whyscream) [inline examples/app.py]: It's no longer plain text

File: mollie/api/objects/order.py [hunk 7]
```diff
@@ -147,6 +146,18 @@ def create_refund(self, data, **params):
         refund = OrderRefunds(self.client).on(self).create(data, **params)
         return refund
 
+    def cancel_lines(self, data=None, **params):
+        """Cancel the lines given. When no lines are given, cancel all the lines.
+
+        We're using the create function,
+        because the API expects a POST request containing the lines you want to cancel.
```
Comment (whyscream) [inline mollie/api/objects/order.py]: This is incorrect. The API expects a DELETE request with a body.

File: mollie/api/objects/order_line.py [hunk 8]
```diff
@@ -107,16 +118,6 @@ def product_url(self):
     def created_at(self):
         return self._get_property('createdAt')
 
-    def cancel(self):
```
Comment (whyscream) [inline mollie/api/objects/order_line.py:110]: This method can still exist, it should call the DELETE with only itself as the argument.

File: tests/test_orders.py [hunk 9]
```diff
@@ -205,3 +206,19 @@ def test_list_order_refund(client, response):
...
     assert_list_object(refunds, Refund)
+
+
+def test_cancel_order_lines(client, response):
+    """Firstly cancel a single line, then cancel all the lines of an order."""
+    response.get('https://api.mollie.com/v2/orders/{order_id}'.format(order_id=ORDER_ID), 'order_single')
+    response.post('https://api.mollie.com/v2/orders/{order_id}/lines'.format(order_id=ORDER_ID), 'empty', 204)
+    order = client.orders.get(ORDER_ID)
+    line = next(order.lines)
+    data = {'lines': []}
```
Comment (whyscream) [inline tests/test_orders.py]: Why are you creating the payload in such a weird way?

**Review by whyscream — CHANGES_REQUESTED** [review]

File: mollie/api/objects/order.py (hunk 7 above)
Comment (capsert) [reply mollie/api/objects/order.py]: Nee, in de comment zei hij dat hij een POST verwachte met de lines die verwijderd moeten worden: `With a POST request body containing the lines.`

File: mollie/api/objects/order_line.py (hunk 8 above)
Comment (whyscream) [reply mollie/api/objects/order_line.py:110]: Correction: we don't have any cancel methods anymore on result objects, so this one should go too.

**Review by whyscream — APPROVED** [review]

**Pull request merged**